src/utils.test.js

```diff
@@ -60,4 +62,71 @@ describe('utils', () => {
       expect(dataHubParams.hash).toBe('#hash1#hash2');
     });
   });
+
+  describe('setInterval custom hook', () => {
+    it('should set an interval and update an outside value', () => {
+      // Arrange
+      jest.useFakeTimers();
+      let counter = 0;
+      renderHook(() =>
+        useInterval({
+          runOnStart: false,
+          delay: 100,
+          callback: () => {
+            counter++;
+          },
+        }),
+      );
+
+      // Act
+      jest.runOnlyPendingTimers();
+
+      // Assert
+      expect(counter).toBe(1);
+    });
+
+    it('should set an interval and pause when delay is null', () => {
+      // Arrange
+      jest.useFakeTimers();
+      let counter = 0;
+      const { rerender } = renderHook(
+        ({ delay }) =>
+          useInterval({
+            runOnStart: false,
+            delay: delay,
+            callback: () => {
+              counter++;
+            },
+          }),
+        { initialProps: { delay: 100 } },
+      );
+      // Act
+      jest.advanceTimersByTime(600);
+      rerender({ delay: null });
+      jest.advanceTimersByTime(800);
+      // Assert
+      expect(counter).toBe(6);
+    });
+
+    it('should set an interval and run callback before start', () => {
+      // Arrange
+      jest.useFakeTimers();
+      let counter = 0;
+      renderHook(() =>
+        useInterval({
+          runOnStart: true,
+          delay: 100,
+          callback: () => {
+            counter++;
+          },
+        }),
+      );
+      // Assert
+      expect(counter).toBe(1);
+      // Act
```

Comment:
I think that in this test is not an _act_ or maybe the `renderHook` is the _act_.

Reply:
Yes I had my doubts here.